From e7f3d2be989959d3f370f37f6cc97a3d6496ef85 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:13:37 -0400
Subject: [alsa] pcm: Warn when buffer preallocation fails

Message-id: <140972823573-31-patchset-jkysela@redhat.com>
Patchwork-id: 90731
O-Subject: [RHEL 7.1 PATCH 030/442] ALSA: PCM: Warn when buffer preallocation fails
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The failures of buffer preallocations at driver initializations aren't
critical but it's still helpful to inform, so that user can know that
something doesn't work as expected.

For example, the recent page allocator change triggered regressions,
but developers didn't notice until recently because the driver didn't
complain.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 6ab08ced64a4d5ea321fd1d5ebbd2217789b040f)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/core/pcm_memory.c b/sound/core/pcm_memory.c
index 01f8eaf..54debc0 100644
--- a/sound/core/pcm_memory.c
+++ b/sound/core/pcm_memory.c
@@ -51,6 +51,7 @@ static const size_t snd_minimum_buffer = 16384;
 static int preallocate_pcm_pages(struct snd_pcm_substream *substream, size_t size)
 {
  struct snd_dma_buffer *dmab = &substream->dma_buffer;
+ size_t orig_size = size;
  int err;
 
  do {
@@ -63,6 +64,10 @@ static int preallocate_pcm_pages(struct snd_pcm_substream *substream, size_t siz
   size >>= 1;
  } while (size >= snd_minimum_buffer);
  dmab->bytes = 0; /* tell error */
+ pr_warn("ALSA pcmC%dD%d%c,%d:%s: cannot preallocate for size %zu\n",
+  substream->pcm->card->number, substream->pcm->device,
+  substream->stream ? 'c' : 'p', substream->number,
+  substream->pcm->name, orig_size);
  return 0;
 }
 
-- 
1.7.1